Index: siesta/Src/Makefile
===================================================================
--- siesta.orig/Src/Makefile	2006-04-19 18:10:49.000000000 +0200
+++ siesta/Src/Makefile	2006-04-19 18:10:49.000000000 +0200
@@ -153,7 +153,7 @@
 	@(cd Libs ; $(MAKE) dc_lapack.a)
 #
 
-version: version.F90
+version.o version_info.mod: version.F90 Makefile arch.make
 	@echo
 	@echo "==> Incorporating information about present compilation (compiler and flags)"
 	@sed "s'SIESTA_ARCH'${SIESTA_ARCH}'g;s'FFLAGS'${FC} ${FFLAGS}'g;s'SIESTA_VERSION'$$(cat ../version.info)'g" version.F90 > compinfo.F90
@@ -162,7 +162,7 @@
 	#@rm -f compinfo.F90
 	@echo
 
-siesta: version $(MPI_INTERFACE) $(NETCDF_INTERFACE) $(FDF) $(WXML) $(XMLPARSER) \
+siesta: version_info.mod $(MPI_INTERFACE) $(NETCDF_INTERFACE) $(FDF) $(WXML) $(XMLPARSER) \
                 $(COMP_LIBS) $(ALL_OBJS) 
 	$(FC) -o siesta \
 	       $(LDFLAGS) $(ALL_OBJS) $(FDF) $(WXML) $(XMLPARSER) $(MPI_INTERFACE)\
@@ -178,7 +178,7 @@
           sys.o timer.o  xml.o m_walltime.o $(SYSOBJ)
 # precision must be the first one here...
 gen-basis.o: $(FDF)
-gen-basis: version $(NETCDF_INTERFACE) $(MPI_INTERFACE) $(FDF) $(WXML) \
+gen-basis: version_info.mod $(NETCDF_INTERFACE) $(MPI_INTERFACE) $(FDF) $(WXML) \
                  $(GEN-BASIS_OBJS)
 	$(FC) -o gen-basis \
 	       $(LDFLAGS) $(GEN-BASIS_OBJS) $(MPI_INTERFACE) $(FDF) $(WXML) \
Index: siesta/Src/arch.make
===================================================================
--- siesta.orig/Src/arch.make	2006-04-19 18:10:49.000000000 +0200
+++ siesta/Src/arch.make	2006-04-19 18:11:18.000000000 +0200
@@ -1,41 +1,63 @@
-SIESTA_ARCH=intel-mkl
 #
-# Intel fortran compiler for linux with mkl optimized blas and lapack
+# This file is part of the SIESTA package.
+#
+# Copyright (c) Fundacion General Universidad Autonoma de Madrid:
+# E.Artacho, J.Gale, A.Garcia, J.Junquera, P.Ordejon, D.Sanchez-Portal
+# and J.M.Soler, 1996-2006.
+#
+# Use of this software constitutes agreement with the full conditions
+# given in the SIESTA license, as signed by all legitimate users.
+#
+SIESTA_ARCH=intel9-mkl8
+#
+# Intel fortran compiler 9 for linux with mkl 8 optimized blas and lapack
 #
 # Be sure to experiment with different optimization options.
 # You have quite a number of combinations to try...
 #
-FC=ifort
+# Note: The -mp1 option is necessary to recover IEEE floating point precision.
 #
-FFLAGS= -w -mp -tpp7 -O3
+FC=ifort
+
+SYS=nag
+
+SP_KIND=4
+DP_KIND=8
+KINDS=$(SP_KIND) $(DP_KIND)
+
+FFLAGS=-w -xN -O3 -mp1
 FFLAGS_DEBUG= -g
-LDFLAGS=-Vaxlib
+LDFLAGS= -static
 COMP_LIBS=
-RANLIB=echo
+RANLIB=ranlib
 #
 NETCDF_LIBS=
 NETCDF_INTERFACE=
 DEFS_CDF=
 #
+#NETCDF_LIBS=$(NNlab_path)/libnetcdf.a
+#NETCDF_INTERFACE=libnetcdf_f90.a
+#DEFS_CDF=-DCDF
+#
 MPI_INTERFACE=
 MPI_INCLUDE=
 DEFS_MPI=
 #
-MKL=/usr/local/intel_mkl_61
+MKL=~/mnt/scratch/intel_mkl_8.0.2
 GUIDE=$(MKL)/lib/32/libguide.a
 LAPACK=$(MKL)/lib/32/libmkl_lapack.a
 BLAS=$(MKL)/lib/32/libmkl_ia32.a
-#G2C=/usr/lib/gcc-lib/i386-redhat-linux/2.96/libg2c.a
-LIBS=$(LAPACK) $(BLAS) $(G2C) $(GUIDE)  -lpthread
-SYS=bsd
-DEFS= $(DEFS_CDF) $(DEFS_MPI)
+LIBS=$(LAPACK) $(BLAS)  $(GUIDE)  -lpthread -lsvml
+DEFS=-DFC_HAVE_FLUSH -DFC_HAVE_ABORT $(DEFS_CDF) $(DEFS_MPI)
 #
 .F.o:
-	$(FC) -c $(FFLAGS)  $(DEFS) $<
+	$(FC) -c $(FFLAGS) $(INCFLAGS)  $(DEFS) $<
 .f.o:
-	$(FC) -c $(FFLAGS)  $<
+	$(FC) -c $(FFLAGS) $(INCFLAGS)   $<
+# cancel implicit Modula-2 rule
+%.o : %.mod
 .F90.o:
-	$(FC) -c $(FFLAGS)  $(DEFS) $<
+	$(FC) -c $(FFLAGS) $(INCFLAGS)  $(DEFS) $<
 .f90.o:
-	$(FC) -c $(FFLAGS)  $<
+	$(FC) -c $(FFLAGS) $(INCFLAGS)   $<
 #
