Index: bibus/bibusStart.py
===================================================================
--- bibus.orig/bibusStart.py	2006-12-15 22:16:09.000000000 +0100
+++ bibus/bibusStart.py	2007-04-18 08:46:35.000000000 +0200
@@ -20,7 +20,12 @@
 import ConfigParser,os,sys
 cp=ConfigParser.ConfigParser()
 fileName = sys.argv[0]
-while os.path.islink(fileName): fileName = os.readlink(fileName)
+while os.path.islink(fileName):
+    l = os.readlink(fileName)
+    if os.path.isabs(l):
+	fileName = l
+    else:
+	fileName = os.path.join(os.path.dirname(fileName), l)
 sourcedir = os.path.abspath( os.path.dirname(fileName) )
 cp.read( os.path.join(sourcedir,'bibus.cfg') )
 python = os.path.abspath( cp.get('PATH','python') )
