Index: matplotlib/lib/matplotlib/axes.py
===================================================================
--- matplotlib.orig/lib/matplotlib/axes.py	2007-07-01 00:15:53.000000000 +0200
+++ matplotlib/lib/matplotlib/axes.py	2007-07-05 15:28:58.000000000 +0200
@@ -2942,13 +2942,20 @@
 
 
         if len(args)==0:
-            handles = []
-            labels = []
+            handles = []; autohandles = []
+            labels = [];  autolabels = []
             for handle in get_handles():
                 label = handle.get_label()
-                if label is not None and label != '' and label[0] != '_':
-                    handles.append(handle)
-                    labels.append(label)
+                if label is not None and label != '':
+                    if label[0] != '_':
+                        handles.append(handle)
+                        labels.append(label)
+                    else:
+                        autohandles.append(handle)
+                        autolabels.append(label[1:])
+            if len(labels) == 0:
+                handles = autohandles
+                labels = autolabels
             loc = popd(kwargs, 'loc', 1)
 
         elif len(args)==1:
