Index: matplotlib/lib/matplotlib/axes.py
===================================================================
--- matplotlib.orig/lib/matplotlib/axes.py	2007-07-19 16:29:25.000000000 +0200
+++ matplotlib/lib/matplotlib/axes.py	2007-07-19 16:47:16.486831750 +0200
@@ -4307,7 +4307,7 @@
                alpha=1.0, vmin=None, vmax=None, origin=None, extent=None)
 
         IMSHOW(X) - plot image X to current axes, resampling to scale to axes
-                    size (X may be numarray/Numeric array or PIL image)
+                    size (X may be NumPy array or PIL image)
 
         IMSHOW(X, **kwargs) - Use keyword args to control image scaling,
         colormapping etc. See below for details
Index: matplotlib/lib/matplotlib/backends/backend_cairo.py
===================================================================
--- matplotlib.orig/lib/matplotlib/backends/backend_cairo.py	2007-07-13 14:36:22.000000000 +0200
+++ matplotlib/lib/matplotlib/backends/backend_cairo.py	2007-07-19 16:47:16.550835750 +0200
@@ -304,11 +304,6 @@
         if _debug: print '%s.%s()' % (self.__class__.__name__, _fn_name())
         # mathtext using the gtk/gdk method
 
-        #if numx.which[0] == "numarray":
-        #   warnings.warn("_draw_mathtext() currently works for numpy, but "
-        #                 "not numarray")
-        #   return
-
         #if not HAVE_CAIRO_NUMPY:
         #    warnings.warn("cairo with Numeric support is required for "
         #                  "_draw_mathtext()")
Index: matplotlib/lib/matplotlib/cm.py
===================================================================
--- matplotlib.orig/lib/matplotlib/cm.py	2007-07-13 14:36:22.000000000 +0200
+++ matplotlib/lib/matplotlib/cm.py	2007-07-19 16:47:16.554836000 +0200
@@ -56,7 +56,7 @@
         return x
 
     def set_array(self, A):
-        'Set the image array from numeric/numarray A'
+        'Set the image array from NumPy array A'
         self._A = A
 
     def get_array(self):
