Index: matplotlib/examples/simple3d.py
===================================================================
--- matplotlib.orig/examples/simple3d.py	2007-06-02 10:54:24.000000000 +0200
+++ matplotlib/examples/simple3d.py	2007-06-02 10:54:37.000000000 +0200
@@ -13,10 +13,10 @@
 
 fig = pylab.gcf()
 
-ax3d = axes3d.Axes3D(fig)
+ax3d = axes3d.Axes3DI(fig)
 plt = fig.axes.append(ax3d)
 
-delta = pi / 199.0
+delta = pi / 99.0
 u = arange(0, 2*pi+(delta*2), delta*2)
 v = arange(0, pi+delta, delta)
 
@@ -24,9 +24,9 @@
 y = outerproduct(sin(u),sin(v))
 z = outerproduct(ones(u.shape), cos(v))
 
-#ax3d.plot_wireframe(x,y,z)
-surf = ax3d.plot_surface(x, y, z)
-surf.set_array(linspace(0, 1.0, len(v)))
+ax3d.plot_wireframe(x,y,z)
+#surf = ax3d.plot_surface(x, y, z)
+#surf.set_array(linspace(0, 1.0, len(v)))
         
 ax3d.set_xlabel('X')
 ax3d.set_ylabel('Y')
