Index: enthought/src/lib/enthought/setup.py
===================================================================
--- enthought.orig/src/lib/enthought/setup.py	2006-11-07 11:16:36.000000000 +0100
+++ enthought/src/lib/enthought/setup.py	2006-11-13 11:09:18.000000000 +0100
@@ -110,7 +110,7 @@
 
     config.add_subpackage('greenlet')
 
-    config.add_subpackage('guitest')
+#     config.add_subpackage('guitest')
 
     config.add_subpackage('help')
     config.add_subpackage('help.tests')
Index: enthought/src/lib/enthought/pyface/tvtk/wxVTKRenderWindowInteractor.py
===================================================================
--- enthought.orig/src/lib/enthought/pyface/tvtk/wxVTKRenderWindowInteractor.py	2006-11-13 14:32:48.000000000 +0100
+++ enthought/src/lib/enthought/pyface/tvtk/wxVTKRenderWindowInteractor.py	2006-11-13 14:32:59.000000000 +0100
@@ -56,7 +56,7 @@
 # This is sometimes necessary under wxGTK or the image is blank.
 # (in wxWindows 2.3.1 and earlier, the GLCanvas had scroll bars)
 try:
-    WX_USE_GL_CANVAS
+    WX_USE_GLCANVAS = 0
 except NameError:
     if wxPlatform == '__WXMSW__':
         WX_USE_GLCANVAS = 0
