Index: cogito/cg-Xlib
===================================================================
--- cogito.orig/cg-Xlib	2006-10-19 02:39:53.000000000 +0200
+++ cogito/cg-Xlib	2006-12-05 16:37:51.000000000 +0100
@@ -246,10 +246,10 @@
 }
 
 # These are shared between cg-diff and cg-log:
-colorify_diffcolors="diffhdr=1;36"
-colorify_diffcolors="$colorify_diffcolors:diffhdradd=1;32:diffadd=32"
-colorify_diffcolors="$colorify_diffcolors:diffhdrmod=1;35:diffmod=35"
-colorify_diffcolors="$colorify_diffcolors:diffhdrrem=1;31:diffrem=31"
+colorify_diffcolors="diffhdr=7"
+colorify_diffcolors="$colorify_diffcolors:diffhdradd=7;32:diffadd=32"
+colorify_diffcolors="$colorify_diffcolors:diffhdrmod=7;35:diffmod=35"
+colorify_diffcolors="$colorify_diffcolors:diffhdrrem=7;31:diffrem=31"
 colorify_diffcolors="$colorify_diffcolors:diffhunk=36:diffctx=34"
 colorify_diffcolors="$colorify_diffcolors:default=0"
 
