Sun Jan  2 13:20:53 CET 2005  Norbert@Nemec-online.de
  * getfinite
Index: numarray/Lib/ieeespecial.py
===================================================================
--- numarray.orig/Lib/ieeespecial.py	2005-07-20 20:17:12.000000000 +0200
+++ numarray/Lib/ieeespecial.py	2005-11-02 11:30:31.000000000 +0100
@@ -347,6 +347,10 @@
     """
     a[isnan(a)]= value
 
+def getfinite(a):
+    """getfinite returns a tuple of indices of 'a' where the values are finite numbers"""
+    return index(a, FINITE)
+
 #def isnan(a):
 #    """Idenitfies elements of 'a' which are NANs, not a number.
 #    """
