Index: numarray/setup.py
===================================================================
--- numarray.orig/setup.py	2005-11-25 17:02:55.000000000 +0100
+++ numarray/setup.py	2006-01-05 14:04:22.000000000 +0100
@@ -227,6 +227,6 @@
 # Do the installation.
 #
 if __name__ == "__main__":
-  if "config" not in sys.argv and ("build" in sys.argv or "install" in sys.argv):
-    sys.argv.insert(sys.argv.index("setup.py")+1, "config")
+#  if "config" not in sys.argv and ("build" in sys.argv or "install" in sys.argv):
+#    sys.argv.insert(sys.argv.index("setup.py")+1, "config")
   main()
