Index: quilt/quilt/push.in
===================================================================
--- quilt.orig/quilt/push.in	2006-02-14 19:31:44.000000000 +0100
+++ quilt/quilt/push.in	2007-05-25 14:35:16.778442250 +0200
@@ -279,6 +279,9 @@
 
 eval set -- "$options"
 
+# defaults:
+[ -t 1 ] && opt_color=1  # --color='auto'
+
 while true
 do
         case "$1" in
Index: quilt/quilt/diff.in
===================================================================
--- quilt.orig/quilt/diff.in	2006-02-14 19:31:44.000000000 +0100
+++ quilt/quilt/diff.in	2007-05-25 14:35:27.575117000 +0200
@@ -136,7 +136,10 @@
 
 eval set -- "$options"
 
+# defaults:
 opt_format=-u
+[ -t 1 ] && opt_color=1  # --color='auto'
+
 while true
 do
 	case "$1" in
