Index: arpack/ARmake.inc
===================================================================
--- arpack.orig/ARmake.inc	1996-04-26 21:12:57.000000000 +0200
+++ arpack/ARmake.inc	2006-08-20 09:13:01.000000000 +0200
@@ -25,14 +25,14 @@
 # | other than your home directory.      |
 # %--------------------------------------%
 #
-home = $(HOME)/ARPACK
+home = $(NNlab)/src/arpack
 #
 #  %--------------------------------------%
 #  | The platform identifier to suffix to |
 #  | the end of library names             |
 #  %--------------------------------------%
 #
-PLAT = SUN4
+#PLAT = Linux
 #
 #  %------------------------------------------------------%
 #  | The directories to find the various pieces of ARPACK |
@@ -43,7 +43,7 @@
 UTILdir      = $(home)/UTIL
 SRCdir       = $(home)/SRC
 #
-DIRS        = $(BLASdir) $(LAPACKdir) $(UTILdir) $(SRCdir)
+# DIRS        = $(BLASdir) $(LAPACKdir) $(UTILdir) $(SRCdir)
 #
 # %-------------------------------------------------------------------%
 # | Comment out the previous line and uncomment the following         |
@@ -51,15 +51,15 @@
 # | NOTE: ARPACK assumes the use of LAPACK version 2 codes.           |
 # %-------------------------------------------------------------------%
 #
-#DIRS         = $(UTILdir) $(SRCdir)
+DIRS         = $(UTILdir) $(SRCdir)
 #
 # %---------------------------------------------------%
 # | The name of the libraries to be created/linked to |
 # %---------------------------------------------------%
 #
-ARPACKLIB  = $(home)/libarpack_$(PLAT).a
-LAPACKLIB = 
-BLASLIB = 
+ARPACKLIB  = $(NNlab)/lib/libarpack.a
+LAPACKLIB = -llapack
+BLASLIB =  -lblas
 #
 ALIBS =  $(ARPACKLIB) $(LAPACKLIB) $(BLASLIB) 
 #
@@ -129,10 +129,3 @@
 ARFLAGS = rv
 #RANLIB  = touch
 RANLIB   = ranlib
-#
-# %----------------------------------%
-# | This is the general help target. |
-# %----------------------------------%
-#
-help:
-	@$(ECHO) "usage: make ?"
